ovh/resource_cloud_project_database_redis_user.go

```diff
@@ -0,0 +1,227 @@
+				Type:        schema.TypeString,
+				Description: "Id of the database cluster",
+				ForceNew:    true,
+				Required:    true,
+			},
+			"categories": {
+				Type:        schema.TypeList,
+				Description: "Categories of the user",
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"channels": {
+				Type:        schema.TypeList,
+				Description: "Channels of the user",
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"commands": {
+				Type:        schema.TypeList,
+				Description: "Commands of the user",
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"keys": {
+				Type:        schema.TypeList,
+				Description: "Keys of the user",
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Name of the user",
+				ForceNew:    true,
+				Required:    true,
+			},
+
+			//Computed
+			"created_at": {
+				Type:        schema.TypeString,
+				Description: "Date of the creation of the user",
+				Computed:    true,
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Description: "Password of the user",
+				Sensitive:   true,
+				Computed:    true,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Description: "Current status of the user",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceCloudProjectDatabaseRedisUserImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	givenId := d.Id()
+	n := 3
+	splitId := strings.SplitN(givenId, "/", n)
+	if len(splitId) != n {
+		return nil, fmt.Errorf("Import Id is not service_name/cluster_id/id formatted")
+	}
+	serviceName := splitId[0]
+	clusterId := splitId[1]
+	id := splitId[2]
+	d.SetId(id)
+	d.Set("cluster_id", clusterId)
+	d.Set("service_name", serviceName)
+
+	results := make([]*schema.ResourceData, 1)
+	results[0] = d
+	return results, nil
+}
+
+func resourceCloudProjectDatabaseRedisUserCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	serviceName := d.Get("service_name").(string)
+	clusterId := d.Get("cluster_id").(string)
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/database/redis/%s/user",
+		url.PathEscape(serviceName),
+		url.PathEscape(clusterId),
+	)
+	params := (&CloudProjectDatabaseRedisUserCreateOpts{}).FromResource(d)
+	res := &CloudProjectDatabaseRedisUserResponse{}
+
+	log.Printf("[DEBUG] Will create user: %+v for cluster %s from project %s", params, clusterId, serviceName)
+	err := config.OVHClient.Post(endpoint, params, res)
+	if err != nil {
+		return fmt.Errorf("calling Post %s with params %+v:\n\t %q", endpoint, params, err)
+	}
+
+	log.Printf("[DEBUG] Waiting for database %s to be READY", clusterId)
+	err = waitForCloudProjectDatabaseReady(config.OVHClient, serviceName, "redis", clusterId, 60*time.Second, 10*time.Second)
```

Comment:
Here you should wait for the user to be properly created waiting for the status to be set at "READY"